src/ReportManager/SchemaForm/SchemaFormContext/index.js

```diff
@@ -0,0 +1,50 @@
+import React, { createContext } from 'react';
+import { FORM_FIELDS_TYPES } from '../constants';
+import { textFieldDetailsFactory } from '../fields/fieldDetailsFactory';
+import { isFieldRequired } from '../utils';
+
+export const SchemaFormContext = createContext(null);
+
+
+const SchemaFormContextProvider = ({ schema, onFormChange, formData, formErrors, children }) => {
+
+  const onFieldChange = (field, value) => {
+    onFormChange({
+      formData: {
+        [field]: value
+      }
+    });
+  };
+
+  const getFieldDetails = (fieldName, schema) => {
+    const fieldSchema = schema.json.properties[fieldName];
+    const isRequired = isFieldRequired(fieldName, schema);
+    const uiDetails = schema.ui.fields[fieldName];
+    const formValue = formData[fieldName] ?? '';
+    const formError = formErrors?.[fieldName] ?? null;
+
+    switch (uiDetails.type) {
+    case FORM_FIELDS_TYPES.TEXT: {
+      return textFieldDetailsFactory(fieldSchema, {
+        ...uiDetails,
+        isRequired
+      }, formValue, formError);
+    }
+
+    default: return {};
+    }
+  };
+
+  const getSchema = () => ({ ...schema });
```

Comment:
Is there a reason to spread the schema instead of just returning it? Remember that spreading just updates the pointers of the top level properties, but the inner properties below `ui` and `json` will still be passed by reference. If you want a cloned object you may want to use a helper like `_.clone`. But is it necessary?

Reply:
There are places where the schema is required, my intention was to expose the schema object without allowing mutation to the original object which is the source of truth, if you agree on this I'll update this method accordingly

Reply:
I supposed it was to protect it from mutation. But again, that will only work for the top level properties of the object "ui" and "json". When you spread, JS creates copies of the first level of the properties, but not for the second level onwards:
```
// Original schema
const schema = {
  json: { /* Deeper properties */ },
  ui: { /* Deeper properties */ },
};

// This method of clonation is "shallow", it copies the content of the first level properties "json" and "ui", but
// the deeper object properties will still be copied by reference (same pointers, meaning that if you modify them
// in "shallowSchemaClone" they will also be modified in the original "schema"!!
const shallowSchemaClone = { ...schema };

// This won't modify the original "schema.json"
shallowSchemaClone.json = { /* New object */ }

// This WILL modify the original "schema.json.properties" because it is copied by reference
shallowSchemaClone.json.properties = { /* New object */ };

// Real clonation at all levels, you can modify any property and it won't be modified in the original "schema"
const deepSchemaClone = _.clone(schema);

// This won't modify the original "schema.json.properties" because it is copied by value
shallowSchemaClone.json.properties = { /* New object */ };
```

Reply:
However, deep cloning can be expensive. If we are going to call this method several times it can have impact on performance because the schemas can go infinitely deep (thanks collections). So, do we really need to clone it? I also like defensive code, but maybe here the best is just to make sure nobody modifies the schema ever (we shouldn't, there's no reason to do so).